### PR TITLE
[Warning] Eliminated AnnotateFormatMethod warnings

### PR DESCRIPTION
**Pull request by hellosagar**
### Overview
Eliminated the [AnnotateFormatMethod](https://errorprone.info/bugpattern/AnnotateFormatMethod) across the entire project.

### Proposed Changes

Added the changes in the following files to fix the error-prone warning
- [Asset.java](https://github.com/robolectric/robolectric/compare/master...hellosagar:annotate-format-method-warning-fix?expand=1#diff-44e9f48ffcff0d656e07849a5a4a98c21efd1bb1751c24aa56fc528392c4dc01)
- [AttributeResolution.java](https://github.com/robolectric/robolectric/compare/master...hellosagar:annotate-format-method-warning-fix?expand=1#diff-b56eec5dd95db47a23a731b3eb1b3586f1e7867917ad542766eb2f5e585512e2)
- [AttributeResolution10.java](https://github.com/robolectric/robolectric/compare/master...hellosagar:annotate-format-method-warning-fix?expand=1#diff-3930f55797eab3fea048baf3c70ec8bcbfdc5518abd4558275ff92f516681946)
- [AttributeResolution9.java](https://github.com/robolectric/robolectric/compare/master...hellosagar:annotate-format-method-warning-fix?expand=1#diff-c4aa6fcdea8db54d4b765b5712ae4fa6c8bc7d118af1d7eb0b7e4953bc385924)
- [ResStringPool.java](https://github.com/robolectric/robolectric/compare/master...hellosagar:annotate-format-method-warning-fix?expand=1#diff-d908defabbb3f2d6f9bc45de445bbc61c223a88e3f1bb7f3bae692a7287c6a39)
- [ResTable.java](https://github.com/robolectric/robolectric/compare/master...hellosagar:annotate-format-method-warning-fix?expand=1#diff-8c7f527ae0e668f2fd8a1eb4f288318bcd63862c70374426fe3307069ceed4b4)
- [ResXMLParser.java](https://github.com/robolectric/robolectric/compare/master...hellosagar:annotate-format-method-warning-fix?expand=1#diff-1d1a02e0e0e0ef4578fccfe8247dcd1a3ef0c42d5403efcfde93f15da323ac30)
- [Util.java](https://github.com/robolectric/robolectric/compare/master...hellosagar:annotate-format-method-warning-fix?expand=1#diff-48bfa6ff48c8f002369075c214ef80a1cf8435ee90bc86d7f63a1dd012ab4f01)

### CODE REVIEW

File: resources/src/main/java/org/robolectric/res/android/Util.java
Comment (utzcoz) [inline resources/src/main/java/org/robolectric/res/android/Util.java:69]: Those code will not change frequently, do we need to include `errorprone` annotations to this files? If I understand correctly, the `AnnotateFormatMethod` warnings are raised by errorprone when building?

File: resources/src/main/java/org/robolectric/res/android/ResXMLParser.java
Comment (utzcoz) [inline resources/src/main/java/org/robolectric/res/android/ResXMLParser.java:38]: @hoisie Could those files can be formatted by google-java-format? If yes, what about formatting it in a single PR and could you help to format those files? I have noticed you have formatted some Java files separately.

File: resources/src/main/java/org/robolectric/res/android/Util.java
Comment (hellosagar) [reply resources/src/main/java/org/robolectric/res/android/Util.java:69]: Yes, we need the error annotation to eliminate the warning, this annotation helps to check if the pattern and value are compatible with each other when passed in the `String.format()`.

Yes, the `AnnotateFormatMethod` warning is raised by the building the project

File: resources/src/main/java/org/robolectric/res/android/ResXMLParser.java
Comment (hellosagar) [reply resources/src/main/java/org/robolectric/res/android/ResXMLParser.java:38]: sorry If I'm mistaken but this [ResXMLParser.java](https://github.com/robolectric/robolectric/pull/7353/files/afd6132ee1dbe36321930399e2692f24fae85768#diff-1d1a02e0e0e0ef4578fccfe8247dcd1a3ef0c42d5403efcfde93f15da323ac30) is no longer needed right as we are going to keep the Kotlin version right?

Comment (utzcoz) [reply resources/src/main/java/org/robolectric/res/android/ResXMLParser.java:38]: Not. We don't want to convert any code released to user/developer to Kotlin now, and only for tests. There are two implementation for resource supporting, legacy one and re-implementation libandroidfw resource logic in Java. This class is used to parse XML of resources in test apk. If you grep `ResXMLParser`, you can find many references to this class.

File: resources/src/main/java/org/robolectric/res/android/Util.java
Comment (utzcoz) [reply resources/src/main/java/org/robolectric/res/android/Util.java:69]: What I mean is related source code will not changed frequently, should we add `FormatMethod` after fixing found problems?

File: resources/src/main/java/org/robolectric/res/android/ResXMLParser.java
Comment (hellosagar) [reply resources/src/main/java/org/robolectric/res/android/ResXMLParser.java:38]: ahh okay, but [ResXMLParser.java](https://github.com/robolectric/robolectric/pull/7353/files/afd6132ee1dbe36321930399e2692f24fae85768#diff-1d1a02e0e0e0ef4578fccfe8247dcd1a3ef0c42d5403efcfde93f15da323ac30) this is not deleted its just formatted according to the google format thingy

File: resources/src/main/java/org/robolectric/res/android/Util.java
Comment (hellosagar) [reply resources/src/main/java/org/robolectric/res/android/Util.java:69]: sorry um.. what do we mean exactly here by `fixing found problems`?

File: resources/src/main/java/org/robolectric/res/android/ResXMLParser.java
Comment (utzcoz) [reply resources/src/main/java/org/robolectric/res/android/ResXMLParser.java:38]: I'm contacting with @hoisie to discuss how to process errorprone fixing brings many formatting changes. It's increase the difficulty to review PR. The formatting problem increase the size of a small PR a lot.

Comment (hellosagar) [reply resources/src/main/java/org/robolectric/res/android/ResXMLParser.java:38]: ohh, yea you are right

File: resources/src/main/java/org/robolectric/res/android/Util.java
Comment (utzcoz) [reply resources/src/main/java/org/robolectric/res/android/Util.java:69]: Oh, what I mean is there are some log formatting problem, such as %p not supported in Java, and we can fix those problems. The source code of this module will not change very frequently, and should we must include errorprone annotations to this module? If we don't include those annotations, will errorprone still raise warnings for us when building?

File: resources/src/main/java/org/robolectric/res/android/ResXMLParser.java
Comment (utzcoz) [reply resources/src/main/java/org/robolectric/res/android/ResXMLParser.java:38]: TBH, the AnnotateFormatMethod warning is lower priority than other code related warnings, and current PR brings errorprone direct dependency(including errorprone code to this module) to resource module, and about six thousand change lines. I'm not very comfortable with current PR.

Comment (hellosagar) [reply resources/src/main/java/org/robolectric/res/android/ResXMLParser.java:38]: Yeah, right. for now I can just push the code with the warning-related fix and leave the formatting changes. and we can keep this PR on hold until I will fix the other significant priority warnings.

File: resources/src/main/java/org/robolectric/res/android/Util.java
Comment (hellosagar) [reply resources/src/main/java/org/robolectric/res/android/Util.java:69]: Yes, if we don't include those the warnings will raise when building the project

File: resources/src/main/java/org/robolectric/res/android/ResXMLParser.java
Comment (hellosagar) [reply resources/src/main/java/org/robolectric/res/android/ResXMLParser.java]: @utzcoz How about if we make it this way, I will make 2 commits, one for warnings and one for refactoring.
During the review, you have to only carefully look at commit 1 and commit 2 you obviously take a look at it too, it's a kind of separation.
Then later before merging, I will squash those two commits into one
In this approach, we have to trust commit 2, it only contains refactoring related things.
but still, it's better than what we are doing now I think.

**Comment by utzcoz** [comment]
@hellosagar Looks new commit to applying google java format generates a reasonable change lines, instead of previous multi thousands lines.

File: resources/src/main/java/org/robolectric/res/android/AttributeResolution10.java
Comment (utzcoz) [inline resources/src/main/java/org/robolectric/res/android/AttributeResolution10.java:97]: Does this format right?

File: resources/src/main/java/org/robolectric/res/android/AttributeResolution10.java
Comment (utzcoz) [inline resources/src/main/java/org/robolectric/res/android/AttributeResolution10.java:163]: Is this one's format right?

**Comment by hellosagar** [comment]
> @hellosagar Look like new commit to applying google java format generates a reasonable change lines, instead of previous multi thousands lines.

Yes.

**Comment by utzcoz** [comment]
There are some formats don't look great, although it has passed checking. And I have wrote some comments for some wrong formats, I think.

**Comment by hellosagar** [comment]
um.. lme check that locally on my system

**Comment by hellosagar** [comment]
@utzcoz you are right, I little surprised how it passed in the CI but mainly [AttributeResolution10.java](https://github.com/robolectric/robolectric/blob/ac9c16b542e9b0db8682f0fe22c92960a95912e1/resources/src/main/java/org/robolectric/res/android/AttributeResolution10.java) contains a lot of weird indentation, Ive reformat it locally, pushing it again

**Comment by utzcoz** [comment]
Why formatted change lines increase a lot again?

**Comment by hellosagar** [comment]
> Why formatted change lines increase a lot again?

because I open that file and pressed the `cmd + ctrl + L ` which performed the google style format as I've their plugin installed.

TDLR: Used the plugin to reformat the file according to the standards
Screenshot 2022-06-11 at 10 00 10 PM

**Comment by utzcoz** [comment]
It's not recommend to add useless formatted lines. Could you format with command in wiki? We just need add formatted lines that affected by this PR.

**Comment by hellosagar** [comment]
Okay, lme undo this

**Comment by hellosagar** [comment]
pushed again according to the wiki, but in that `ALGOI` method has a weird indentation

**Comment by utzcoz** [comment]
Could you fix them manually and check locally whether they can pass google format checking?

**Comment by hellosagar** [comment]
> Could you fix them manually one by one and check locally whether they can pass google format checking?

I tried before pushing this 😅, but it shows a google format error.

**Comment by utzcoz** [comment]
Okay, I will check and verify it tomorrow.

**Comment by utzcoz** [comment]
@hellosagar could you rebase and squash commits for this PR? I think wrong formatting are generated by wrong HEAD when using google-java-format locally. We have analyzed this problem before. Maybe you can get correct code formatting after rebasing, squashing, and running google-java-format with correct command locally again.

**Comment by hellosagar** [comment]
I've ran the formatting locally and pushed it again

File: resources/src/main/java/org/robolectric/res/android/AttributeResolution10.java
Comment (utzcoz) [reply resources/src/main/java/org/robolectric/res/android/AttributeResolution10.java:97]: It's too strange that this wrong formatting existing.

**Comment by hellosagar** [comment]
@utzcoz any update?

**Comment by utzcoz** [comment]
@hellosagar Please fix incorrect format problem and rebase it to latest master.

**Comment by hellosagar** [comment]
:integration_tests:androidx_test:connectedDebugAndroidTest test is even failing in master as well

Below is the log generated from `master` branch

image

File: resources/src/main/java/org/robolectric/res/android/AttributeResolution10.java
Comment (utzcoz) [inline resources/src/main/java/org/robolectric/res/android/AttributeResolution10.java:350]: Please fix these unwanted and incorrect code formatting.

**Review by utzcoz — CHANGES_REQUESTED** [review]

File: resources/src/main/java/org/robolectric/res/android/AttributeResolution10.java
Comment (hellosagar) [reply resources/src/main/java/org/robolectric/res/android/AttributeResolution10.java:350]: Sorry, I'll check again

**Comment by hellosagar** [comment]
@utzcoz I've added a separate commit for easy review, later I'll squash it. Please have a look at it

**Comment by utzcoz** [comment]
> @utzcoz I've added a separate commit for easy review, later I'll squash it. Please have a look at it

You can squash it. Hi @hoisie What about skipping format error for this special case?

**Comment by hellosagar** [comment]
Squashed :)

File: resources/src/main/java/org/robolectric/res/android/Util.java
Comment (utzcoz) [inline resources/src/main/java/org/robolectric/res/android/Util.java:64]: Looks like only `FormatMethod` is needed to fix `AnnotateFormatMethod` warnings. Could we reset above ALOGI/D/V changes?

Comment (hellosagar) [reply resources/src/main/java/org/robolectric/res/android/Util.java:64]: But `ALOGI/D/V` methods already have `@FormatMethod` annotation

- ALOGI: https://github.com/robolectric/robolectric/pull/7353/files/2b875a8c788b9f99401768154b97f28907622208#diff-48bfa6ff48c8f002369075c214ef80a1cf8435ee90bc86d7f63a1dd012ab4f01R76-R77
- ALOGV: https://github.com/robolectric/robolectric/pull/7353/files/2b875a8c788b9f99401768154b97f28907622208#diff-48bfa6ff48c8f002369075c214ef80a1cf8435ee90bc86d7f63a1dd012ab4f01R69-R70
- ALOGD: https://github.com/robolectric/robolectric/pull/7353/files/2b875a8c788b9f99401768154b97f28907622208#diff-48bfa6ff48c8f002369075c214ef80a1cf8435ee90bc86d7f63a1dd012ab4f01R57-R58

Comment (utzcoz) [reply resources/src/main/java/org/robolectric/res/android/Util.java:64]: Could we only mark ALOGI/D/V method with `FormatMethod` and don't change ALOGI/D/V usage?

Comment (hellosagar) [reply resources/src/main/java/org/robolectric/res/android/Util.java:64]: removed the changes from the calling place of `ALOGI/D/V` usages

**Comment by utzcoz** [comment]
@hellosagar Could you enable `kDebugStringPoolNoisy` and build and run tests locally to check whether your changes can work without error? The default value of `kDebugStringPoolNoisy` is false, and I think current CI will not run into some code you modified.

**Comment by hellosagar** [comment]
Locally all checks are passing, without enabling the `kDebugStringPoolNoisy` in `ResTable` and `ResXMLParser`

Screenshot from 2022-09-17 11-55-22


and this is the output after enabling the `kDebugStringPoolNoisy` in `ResTable` and `ResXMLParser`

Screenshot from 2022-09-17 12-08-35

**Comment by utzcoz** [comment]
Hi @hoisie , because there are strange code formatting generated by google-java-format, so I leave format checking error here, and merge this PR firstly.

**Pull request merged**
